services/device-registry-mongodb/src/test/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationServiceTest.java

```diff
@@ -0,0 +1,121 @@
+import org.eclipse.hono.service.registration.AbstractRegistrationServiceTest;
+import org.eclipse.hono.service.registration.RegistrationService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.opentracing.noop.NoopSpan;
+import io.vertx.core.Context;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.mongo.IndexOptions;
+import io.vertx.ext.mongo.MongoClient;
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+@ExtendWith(VertxExtension.class)
+@Timeout(timeUnit = TimeUnit.SECONDS, value = 3)
+class MongoDbBasedRegistrationServiceTest extends AbstractRegistrationServiceTest {
+
+    private final String DUMMY_TENANT_ID = "dummyTenantId";
+    private final String DUMMY_DEVICE_ID = "dummyDeviceId";
+    private final String DUMMY_VERSION = "dummyVersion";
+    private MongoDbCallExecutor mongoDbCallExecutor;
+    private MongoClient mongoClient;
+    private MongoDbBasedRegistrationService svc;
+
+    @BeforeEach
+    void setUp() {
+        final Context ctx = mock(Context.class);
+        final Vertx vertx = mock(Vertx.class);
+        final MongoDbBasedRegistrationConfigProperties config = new MongoDbBasedRegistrationConfigProperties();
+        svc = spy(new MongoDbBasedRegistrationService());
+        svc.setConfig(config);
+        svc.init(vertx, ctx);
+
+        mongoDbCallExecutor = mock(MongoDbCallExecutor.class);
+        mongoClient = new MongoDbClientMock();
+        prepareMongoDbCallExecutorMock();
+        svc.setExecutor(mongoDbCallExecutor);
+        svc.start(Promise.promise());
+    }
+
+    private void prepareMongoDbCallExecutorMock() {
+        when(mongoDbCallExecutor.getMongoClient()).thenReturn(mongoClient);
+        when(mongoDbCallExecutor.createCollectionIndex(anyString(), any(JsonObject.class), any(IndexOptions.class),
+                anyInt()))
+                        .then(invocation -> {
+                            final String collection = invocation.getArgument(0);
+                            final JsonObject key = invocation.getArgument(1);
+                            final IndexOptions options = invocation.getArgument(2);
+                            final Promise<Void> indexCreated = Promise.promise();
+                            mongoClient.createIndexWithOptions(collection, key, options, indexCreated);
+                            return indexCreated;
+                        });
+    }
+
+    @Override
+    public RegistrationService getRegistrationService() {
+        return svc;
+    }
+
+    @Override
+    public DeviceManagementService getDeviceManagementService() {
+        return svc;
+    }
+
+    @Test
+    public void testDeleteDeviceWithResourceVersionOfNonExistingDeviceFails(final VertxTestContext ctx) {
```

Comment:
From the test name, it seems to me that it is not specific to the _MongoDB_ based implementation and it is applicable in general to all implementations. FMPOV it is better to pull this test up to the `AbstractRegistrationServiceTest`.

Reply:
done